tensorflow_addons/optimizers/moving_average.py

```diff
@@ -50,6 +50,9 @@ def __init__(self,
 
         super(MovingAverage, self).__init__(name, **kwargs)
 
+        if isinstance(optimizer, str):
```

Comment:
Should we really need to add this condition? Maybe we could let [tf.keras.optimizer.get](https://github.com/tensorflow/tensorflow/blob/master/tensorflow/python/keras/optimizers.py#L820-L850) check this implicitly.

Reply:
If the user passes an invalid arg to `optimizer`, the error message is less obvious:
```
ValueError('Could not interpret optimizer identifier:', identifier)
```
So I decided to do a manual check with more obvious error messages. What do you think?